homeassistant/components/fritzbox_callmonitor/sensor.py

```diff
@@ -1,114 +1,137 @@
+    INTERVAL_RECONNECT,
+    LOGGER,
+    MANUFACTURER,
+    STATE_DIALING,
+    STATE_IDLE,
+    STATE_RINGING,
+    STATE_TALKING,
+)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_PASSWORD, default="admin"): cv.string,
-        vol.Optional(CONF_USERNAME, default=""): cv.string,
-        vol.Optional(CONF_PHONEBOOK, default=0): cv.positive_int,
-        vol.Optional(CONF_PREFIXES, default=[]): vol.All(cv.ensure_list, [cv.string]),
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PHONEBOOK, default=DEFAULT_PHONEBOOK): cv.positive_int,
+        vol.Optional(CONF_PREFIXES): vol.All(cv.ensure_list, [cv.string]),
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up Fritz!Box call monitor sensor platform."""
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    # Try to resolve a hostname; if it is already an IP, it will be returned as-is
-    try:
-        host = socket.gethostbyname(host)
-    except OSError:
-        _LOGGER.error("Could not resolve hostname %s", host)
-        return
-    port = config.get(CONF_PORT)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    phonebook_id = config.get(CONF_PHONEBOOK)
-    prefixes = config.get(CONF_PREFIXES)
-
-    try:
-        phonebook = FritzBoxPhonebook(
-            host=host,
-            port=port,
-            username=username,
-            password=password,
-            phonebook_id=phonebook_id,
-            prefixes=prefixes,
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import the platform into a config entry."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
         )
-    except:  # noqa: E722 pylint: disable=bare-except
-        phonebook = None
-        _LOGGER.warning("Phonebook with ID %s not found on Fritz!Box", phonebook_id)
-
-    sensor = FritzBoxCallSensor(name=name, phonebook=phonebook)
-
-    add_entities([sensor])
-
-    monitor = FritzBoxCallMonitor(host=host, port=port, sensor=sensor)
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the fritzbox_callmonitor sensor from config_entry."""
+    phonebook = hass.data[DOMAIN][config_entry.entry_id]
+
+    phonebook_id = config_entry.data[CONF_PHONEBOOK]
+    prefixes = config_entry.data[CONF_PREFIXES]
+    host = config_entry.data[CONF_HOST]
+    port = config_entry.data[CONF_PORT]
+
+    phonebook_info = await hass.async_add_executor_job(
+        partial(phonebook.fph.phonebook_info, phonebook_id)
```

Comment:
No need for partial unless you're using keyword arguments.
```suggestion
        phonebook.fph.phonebook_info, phonebook_id
```